src/kademlia/dht_storage.cpp

```diff
@@ -565,13 +574,26 @@ namespace
 					++i;
 			}
 		}
+
+		void remove_least_important_immutable_item()
+		{
+			//HELP HERE!
```

Comment:
I think a reasonable start would be to take the code from immutable_item_comparator::operator() and put it in here. In fact, this logic should probably be identical for immutable and mutale (as steven pointed out). Perhaps these two functions could be a single one, that picks the best candidate for removal from a container (in order to work for both mutable and immutable, it may have to be a template). These functions aren't part of the interface though, right? They aren't meant to be virtual functions.

Basically what immutable_item_comparator::operator() does is that it looks at the distance of an item from our node ID, weighing it together with the number of unique IPs that have announce it (which is meant to represent popularity) and compare those. So, it prioritizes items close to our ID, because those are the ones we're supposed to store, but it also has an affinity to store popular items, to make it harder to attack and overwrite popular items.

In the world of multiple node IDs, instead of calculating the distance to our node ID, it would have to take the minimum of all the distances to all of our node IDs.